src/tasks/Microsoft.NET.Sdk.WebAssembly.Pack.Tasks/ComputeWasmBuildAssets.cs

```diff
@@ -0,0 +1,268 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Microsoft.NET.Sdk.WebAssembly;
+
+namespace Microsoft.NET.Sdk.WebAssembly
+{
+    // This task does the build work of processing the project inputs and producing a set of pseudo-static web assets
+    // specific to Blazor.
+    public class ComputeWasmBuildAssets : Task
+    {
+        [Required]
+        public ITaskItem[] Candidates { get; set; }
+
+        public ITaskItem CustomIcuCandidate { get; set; }
+
+        [Required]
+        public ITaskItem[] ProjectAssembly { get; set; }
+
+        [Required]
+        public ITaskItem[] ProjectDebugSymbols { get; set; }
+
+        [Required]
+        public ITaskItem[] SatelliteAssemblies { get; set; }
+
+        [Required]
+        public ITaskItem[] ProjectSatelliteAssemblies { get; set; }
+
+        [Required]
+        public string OutputPath { get; set; }
+
+        [Required]
+        public bool TimeZoneSupport { get; set; }
+
+        [Required]
+        public bool InvariantGlobalization { get; set; }
+
+        [Required]
+        public bool CopySymbols { get; set; }
+
+        public bool FingerprintDotNetJs { get; set; }
+
+        [Output]
+        public ITaskItem[] AssetCandidates { get; set; }
+
+        [Output]
+        public ITaskItem[] FilesToRemove { get; set; }
+
+        public override bool Execute()
```

Comment:
nit: I would suggest renaming this to `ExecuteInternal`, called by `Execute` which has:

https://github.com/dotnet/runtime/blob/19ff97834663497ff6fd5207f6f64bbe2af93cba/src/tasks/WasmAppBuilder/ManagedToNativeGenerator.cs#L55-L64

This would make the main method simpler, and allow use of
```csharp
throw new LogAsErrorException("Invalid number of project assemblies '{0}'", string.Join("," + Environment.NewLine, ProjectAssembly.Select(a => a.ItemSpec)));
```

Reply:
I can do it in follow-up PR. I would like to focus on the functional part at the moment